src/topology/algebra/valuation.lean

```diff
@@ -13,29 +13,34 @@ import ring_theory.valuation.basic
 
 In this file, we define the non archimedean topology induced by a valuation on a ring.
 The main definition is a `valued` type class which equips a ring with a valuation taking
-values in a group with zero (living in the same universe). Other instances are then deduced from
-this.
+values in a group with zero. Other instances are then deduced from this.
 -/
 
 open_locale classical topological_space
 open set valuation
 noncomputable theory
 
-universe u
+universes v u
 
-/-- A valued ring is a ring that comes equipped with a distinguished valuation.-/
-class valued (R : Type u) [ring R] :=
-(Γ₀ : Type u)
-[grp : linear_ordered_comm_group_with_zero Γ₀]
-(v : valuation R Γ₀)
+/-- A valued ring is a ring that comes equipped with a distinguished valuation. The class `valued`
+is designed for the situation that there is a canonical valuation on the ring. It allows such a
+valuation to be registered as a typeclass; this is used for instance by `valued.topological_space`.
 
-attribute [instance] valued.grp
+TODO: show that there always exists an equivalent valuation taking values in a type belonging to
+the same universe as the ring. -/
+class valued (R : Type u) [ring R] (Γ₀ : out_param (Type v))
+  [linear_ordered_comm_group_with_zero Γ₀] :=
+(v : valuation R Γ₀)
 
 namespace valued
-variables {R : Type*} [ring R] [valued R]
+variables {R : Type u} [ring R] (Γ₀ : Type v) [linear_ordered_comm_group_with_zero Γ₀]
+  [hv : valued R Γ₀]
+
+include hv
 
 /-- The basis of open subgroups for the topology on a valued ring.-/
-lemma subgroups_basis : ring_subgroups_basis (λ γ : (Γ₀ R)ˣ, valued.v.lt_add_subgroup γ) :=
+lemma subgroups_basis :
+  ring_subgroups_basis (λ γ : Γ₀ˣ, (valued.v.lt_add_subgroup γ : add_subgroup R)) :=
```

Comment:
This one is tempting to change to 
```suggestion
lemma _root_.valuation.subgroups_basis (v : valuation R Γ₀) :
  ring_subgroups_basis (λ γ : Γ₀ˣ, (v.lt_add_subgroup γ : add_subgroup R)) :=
```
but that's probably out of scope for this PR.